### PR TITLE
Replace kokkos c++11 impl

### PR DESCRIPTION
**Pull request by masterleinad**
@trilinos/PyTrilinos @trilinos/intrepid2 @trilinos/minitensor @trilinos/muelu @trilinos/shylu @trilinos/stokhos @trilinos/teuchos @trilinos/tpetra @trilinos/trilinoscouplings @trilinos/tpetra

## Motivation
https://github.com/kokkos/kokkos/pull/2679 and https://github.com/kokkos/kokkos/pull/2670 remove `Kokkos::Impl::is_same` and `Kokkos::Impl::enable_if` so the won't be available in releases after 3.0.

## Testing
Building Trilinos still works for me. Only compile-time errors due to missing include files are possible.

### CODE REVIEW

**Comment by lucbv** [comment]
@masterleinad unfortunately if you are not a trilinos organization member, you cannot @ mention teams... I would recommend that you set the labels instead. In the meantime I'll do the team mentions for you:
@trilinos/PyTrilinos @trilinos/intrepid2 @trilinos/minitensor @trilinos/muelu @trilinos/shylu @trilinos/stokhos @trilinos/teuchos @trilinos/tpetra @trilinos/trilinoscouplings @trilinos/tpetra

**Comment by masterleinad** [comment]
> @masterleinad unfortunately if you are not a trilinos organization member, you cannot @ mention teams... I would recommend that you set the labels instead

I know, I am never sure if I should do that anyway... I can't set labels either.

**Comment by masterleinad** [comment]
Hmm... the output for the failing tests is not really telling. I tried to match the failing configuration as good as I could but don't see the build failing. I am happy to look into this some more if someone can extract the real compile-time errors.

File: packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp [hunk 1]
```diff
@@ -207,20 +213,22 @@ void Trilinos_Util_ReadHpc2Epetra_internal(
...
         (*b)[cur_local_row] = bt;
         (*xexact)[cur_local_row] = xxt;
       } else {
-#ifndef NDEBUG
-        cnt =
-#endif
+//#ifndef NDEBUG
+         auto cnt =
+//#endif
         fscanf(in_file, "%lf %lf %lf",vp, vp, vp); // or thrown away
```
Comment (mhoemmen) [inline packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: You should just be able to write `(void) fscanf(...);`.

Comment (masterleinad) [reply packages/triutils/src/Trilinos_Util_ReadHpc2Epetra.cpp]: Oh, this should not have been in this pull request... Fixed now.

**Comment by ndellingwood** [comment]
This showed up in the failing intrepid2 test ` Intrepid2_unit-test_Discretization_Basis_HierarchicalBases_Hierarchical_Basis_Tests_MPI_1` on Ride:

```
ibv_exp_query_device: invalid comp_mask !!! (comp_mask = 0x400000000 valid_mask = 0x1)
[ride10][[18477,1],0][btl_openib_component.c:1648:init_one_device] error obtaining device attributes for mlx5_1 errno says Invalid argument
```

This shouldn't occur due to the changes in this PR, adding RETEST label.

**Comment by masterleinad** [comment]
Thanks!

**Pull request merged**